pydm/data_plugins/epics_plugins/pyepics_plugin_component.py

```diff
@@ -1,165 +1,98 @@
-        self.add_listener(channel)
-
-        self._value = None
-        self._severity = None
-        self._precision = None
-        self._enum_strs = None
-        self._unit = None
-        self._upper_ctrl_limit = None
-        self._lower_ctrl_limit = None
-
-    def clear_cache(self):
-        self._value = None
-        self._severity = None
-        self._precision = None
-        self._enum_strs = None
-        self._unit = None
-        self._upper_ctrl_limit = None
-        self._lower_ctrl_limit = None
-
-    def send_new_value(self, value=None, char_value=None, count=None, ftype=None, type=None, *args, **kws):
+
+    def send_new_value(self, value=None, **kws):
         self.update_ctrl_vars(**kws)
 
-        if value is not None and not np.array_equal(value, self._value):
-            self._value = value
-            if isinstance(value, np.ndarray):
-                self.new_value_signal[np.ndarray].emit(value)
-            else:
-                if ftype in int_types:
-                    try:
-                        self.new_value_signal[int].emit(int(value))
-                    except ValueError:  # This happens when a string is empty
-                        # HACK since looks like for PyEpics a 1 element array
-                        # is in fact a scalar. =( I will try to address this
-                        # with Matt Newville
-                        self.new_value_signal[str].emit(char_value)
-                elif ftype in float_types:
-                    self.new_value_signal[float].emit(float(value))
-                else:
-                    self.new_value_signal[str].emit(char_value)
-
-    def update_ctrl_vars(self, units=None, enum_strs=None, severity=None, upper_ctrl_limit=None, lower_ctrl_limit=None, precision=None, *args, **kws):
-        if severity is not None and self._severity != severity:
-            self._severity = severity
-            self.new_severity_signal.emit(int(severity))
-        if precision is not None and self._precision != precision:
-            self._precision = precision
-            self.prec_signal.emit(precision)
-        if enum_strs is not None and self._enum_strs != enum_strs:
-            self._enum_strs = enum_strs
+        if value is not None:
+            self.data[DataKeys.VALUE] = value
+
+        self.send_to_channel()
+
+    def update_ctrl_vars(self, units=None, enum_strs=None, severity=None,
+                         upper_ctrl_limit=None, lower_ctrl_limit=None,
+                         precision=None, *args, **kws):
+        if severity is not None:
+            self.data[DataKeys.SEVERITY] = severity
+        if precision is not None:
+            self.data[DataKeys.PRECISION] = precision
+        if enum_strs is not None:
             try:
-                enum_strs = tuple(b.decode(encoding='ascii') for b in enum_strs)
+                enum_strs = tuple(
+                    b.decode(encoding='ascii') for b in enum_strs)
```

Comment:
I thought pyepics hands back strings here?

Reply:
This was introduced at https://github.com/slaclab/pydm/commit/d94c5ab76930bdad8c1132b0d9f86bc4d9db9054 .
I'm not sure why but maybe @mattgibbs has some memory of why that was needed...